src/usermodel.cpp

```diff
@@ -463,138 +495,52 @@ void UserModel::setGuestEnabled(bool enabled)
+        if (!enabled) {
             m_transitioning.remove(SAILFISH_USERMANAGER_GUEST_UID);
             auto idx = index(m_uidsToRows.value(SAILFISH_USERMANAGER_GUEST_UID), 0);
             emit dataChanged(idx, idx, QVector<int>() << TransitioningRole);
         }
-    } // else wait for signals
-    call->deleteLater();
+    });
 }
 
 void UserModel::createInterface()
 {
     if (!m_dBusInterface) {
         qCDebug(lcUsersLog) << "Creating interface to user-managerd";
-        m_dBusInterface = new QDBusInterface(UserManagerService, UserManagerPath, UserManagerInterface,
-                                             QDBusConnection::systemBus(), this);
-        connect(m_dBusInterface, SIGNAL(userAdded(const SailfishUserManagerEntry &)),
-                this, SLOT(onUserAdded(const SailfishUserManagerEntry &)), Qt::QueuedConnection);
-        connect(m_dBusInterface, SIGNAL(userModified(uint, const QString &)),
-                this, SLOT(onUserModified(uint, const QString &)));
-        connect(m_dBusInterface, SIGNAL(userRemoved(uint)),
-                this, SLOT(onUserRemoved(uint)));
-        connect(m_dBusInterface, SIGNAL(currentUserChanged(uint)),
-                this, SLOT(onCurrentUserChanged(uint)));
-        connect(m_dBusInterface, SIGNAL(currentUserChangeFailed(uint)),
-                this, SLOT(onCurrentUserChangeFailed(uint)));
-        connect(m_dBusInterface, SIGNAL(guestUserEnabled(bool)),
-                this, SLOT(onGuestUserEnabled(bool)));
+        m_dBusInterface = new NemoDBus::Interface(
+                this, QDBusConnection::systemBus(),
+                UserManagerService, UserManagerPath, UserManagerInterface);
+        m_dBusInterface->connectToSignal(
+                QStringLiteral("userAdded"),
+                SLOT(onUserAdded(const SailfishUserManagerEntry &)));
+        m_dBusInterface->connectToSignal(
+                QStringLiteral("userModified"),
+                SLOT(onUserModified(uint, const QString &)));
+        m_dBusInterface->connectToSignal(
+                QStringLiteral("userRemoved"),
+                SLOT(onUserRemoved(uint)));
+        m_dBusInterface->connectToSignal(
+                QStringLiteral("currentUserChanged"),
+                SLOT(onCurrentUserChanged(uint)));
+        m_dBusInterface->connectToSignal(
+                QStringLiteral("currentUserChangeFailed"),
+                SLOT(onCurrentUserChangeFailed(uint)));
+        m_dBusInterface->connectToSignal(
+                QStringLiteral("guestUserEnabled"),
+                SLOT(onGuestUserEnabled(bool)));
     }
 }
 
 void UserModel::destroyInterface() {
     if (m_dBusInterface) {
         qCDebug(lcUsersLog) << "Destroying interface to user-managerd";
-        disconnect(m_dBusInterface, 0, this, 0);
-        m_dBusInterface->deleteLater();
+        delete m_dBusInterface; // TODO: will this also disconnect the signals?
```

Comment:
@adenexter wondering about this one, I guess internally the connection is from the QDBusConnection to the QObject, rather than from the NemoDBus::Interface, so maybe deleting the interface here won't disconnect signals appropriately.  Or am I just confusing myself?

Reply:
It won't. Interface::connectToSignal() is just QDBusConnection::connect() with fewer arguments.

Reply:
Thanks. Manually added disconnects now.